### DISCUSSION TITLE
What is the standard way to get back to the table object from a column object?

### DISCUSSION THREAD

**Discussion by sprivite** [ibis-project/ibis#10569] [Q&A] [answered]
I want to pass a reference to a column in a function and have that function still be able to access the underlying table. 

Through this discussion: 
https://github.com/ibis-project/ibis/discussions/7248

I ended up concluding that I can access that parent through the following:

table.COLUMN_NAME.op().rel.parent.name

but this is just the *name* of the table, not the table object itself. And also it's just unclear to me if this will always work.

Any tips?

[For context, I'm building a wrapper of ibis for the particular data structure we work with most commonly. I have tables X, Y and Z and the library knows how to join these tables. I want to pass a filter conditions on Y and Z and return rows on X that satisfy (after appropriately joining) the conditions. So I need to be able to go from filter conditions or columns back to source tables. Can give you more details if you'd like.]

**Comment by cpcloud** [comment, marked as answer]
`.rel` should be sufficient there to get the table operation, and for any operation, you can turn it into an expression by calling `.to_expr()` on it.

So `table.COLUMN_NAME.op().rel.to_expr()` should I think be what you want.

**Reply by sprivite** [reply]
Ah thank you so much, this is exactly what I was looking for!!
